### PR TITLE
Refactored and documented PWM and OutputCompare Modules

### PR DESCRIPTION
**Pull request by sergei1152**
**Changes**:
- Documented OutputCompare and PWM modules
- Moved out Timer2 logic into the timer module (such as setting Timer2 period, etc)
- Moved the TImer2 initialization call to the PWM module (used to be in InputCompare)
- Removed any unused functions from PWM, InputCompare, and OutputCompare modules
- renamed and moved around the define statements to make more sense

Note that this is only a refactor, however hasn't been tested yet. I'll test this afterwords with an integration test

### CODE REVIEW

File: Autopilot/AttitudeManager/PWM.h [hunk 1]
```diff
@@ -1,150 +1,96 @@
...
- *                                  For instance, if the range is from -1024 to 1024,
- *                                  the scale factor would be (half the desired output range)/1024
- *        unsigned int signalOffset -> Compensation for a non-zero offset on the input. This is
- *                                      known as trim.
- *
- * Output: None.
- *
- *****************************************************************************/
-void PWMOutputCalibration(unsigned int channel, float signalScaleFactor, unsigned int signalOffset);
+#define SP_RANGE MAX_PWM//(UPPER_PWM - MIDDLE_PWM)
```
Comment (chrishajduk84) [inline Autopilot/AttitudeManager/PWM.h]: Is this still correct?  From what I recall SP_RANGE was indeed half of the actual period.... But  I don't remember fully.

Comment (chrishajduk84) [inline Autopilot/AttitudeManager/PWM.h]: Consider changing this variable name,  since you are in the process of Refactoring.

Comment (sergei1152) [inline Autopilot/AttitudeManager/PWM.h]: I've actually got no idea what this variable does, hence I didnt document it or change it. Here is where its used:

In orientation control.h:
```
#define THROTTLE_SCALE_FACTOR SP_RANGE///Remove this * 2 if having problems
#define FLAP_SCALE_FACTOR SP_RANGE
```
In cameraManager.h
//Constants
```
#define GIMBAL_PWM_RANGE SP_RANGE //Range one way (IE. Positive or Negative range, not both)
```
And in AttitudeManager:
```
int getPitchAngleInput(char source){
    if (source == PITCH_RC_SOURCE){
        return (int)((input_RC_PitchRate / ((float)SP_RANGE / MAX_PITCH_ANGLE) ));
    }
    else if (source == PITCH_GS_SOURCE){
        return input_GS_Pitch;
    }
    else
        return 0;
}
```

If you could shed some light as to what this does, that'd be good. I can document the variable after

Comment (chrishajduk84) [inline Autopilot/AttitudeManager/PWM.h]: Its supposed to just be a scaling factor for half the PWM range. Sort of similar to how you typically convert between deg and rad using 180/pi.

I guess an appropriate definition would be the PWM half range.
Which in terms of readability is better than just "MAX_PWM".

The actual appropriate definition should probably be: (MAX_PWM - MIN_PWM)/2
Which in this case (and probably all cases) is equal to 1024.

**Pull request merged**